### DISCUSSION TITLE
How to bulk update splitAtom?

### DISCUSSION THREAD

**Discussion by zolboo-gtn** [pmndrs/jotai#2500] [Q&A] [answered]
There is recursive large data structured like this
```
{
  ...,
  "children": [
    ...,
    {
      children: [
        ...
      ]
    }
  ]
}
```

and i am using splitAtom to control and render the children.
```
  const childAtomsAtom = splitAtom(childrenAtom);
  const [childAtoms] = useAtom(childAtomsAtom);
```
I need to change the order of the children at a deep layer and for that i am using framer-motion.
[https://www.framer.com/motion/reorder](https://www.framer.com/motion/reorder)
```
<Reorder.Group values={childAtoms} onReorder={(reorderedAtoms) => {}}>
  {childAtoms.map((childAtom) => (
    <Reorder.Item
      key={`${childAtom}`}
      nodeAtom={childAtom}
    >
      ...
    </Reorder.Item>
  ))}
</Reorder.Group>;

```
`onReorder` callback returns array of reordered atoms.

Is there a way to directly update `childAtoms` in bulk like `setChildAtoms(reorderedAtoms)`?
Or is it considered bad practice?

**Comment by dai-shi** [comment, marked as answer]
Good point. I thinks reordering at once is unspported at the moment.
https://github.com/pmndrs/jotai/blob/9ceaef6403a9d932590b37852734bd8bd4958b74/src/vanilla/utils/splitAtom.ts#L28-L39
I'm not sure if it's common practice, but it seems reasonable to support.
You or anyone would be welcome to give a try.

**Reply by zolboo-gtn** [reply]
I see, thank you for the reply.

**Comment by zolboo-gtn** [comment]
Would it suffice?
https://github.com/pmndrs/jotai/pull/2502
